http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala

```diff
@@ -516,6 +523,150 @@ class AzurePubsubHandlerInterp[F[_]: Parallel](
+              )
+            )
+          } yield ()
+        case JobReport.StatusEnum.RUNNING =>
+          params.diskId.traverse(id =>
+            dbRef.inTransaction(persistentDiskQuery.updateStatus(id, DiskStatus.Error, ctx.now))
+          ) >> F.raiseError[Unit](
+            AzureRuntimeDeletionError(
+              params.runtime.id,
+              params.workspaceId,
+              s"Wsm deleteDisk was not completed within ${config.deleteDiskPollConfig.maxAttempts} attempts with ${config.deleteDiskPollConfig.interval} delay, disk resource id ${params.wsmResourceId}"
+            )
+          )
+        case JobReport.StatusEnum.SUCCEEDED =>
+          for {
+            _ <- logger.info(ctx.loggingCtx)(
+              s"disk  for runtime ${params.runtime.id} with resource id ${params.wsmResourceId} is deleted successfully"
+            )
+          } yield ()
+      }
+    } yield ()
+
+  private def monitorDeleteVm(params: PollVmParams)(implicit
+    ev: Ask[F, AppContext]
+  ): F[Unit] =
+    for {
+      ctx <- ev.ask
+      wsmApi <- buildWsmControlledResourceApiClient
+      getWsmJobResult = F.delay(wsmApi.getDeleteAzureVmResult(params.workspaceId.value, params.jobId.value))
+
+      _ <- F.sleep(
+        config.deleteVmPollConfig.initialDelay
+      )
+
+      resp <- streamFUntilDone(
+        getWsmJobResult,
+        config.deleteVmPollConfig.maxAttempts,
+        config.deleteVmPollConfig.interval
+      ).compile.lastOrError
+
+      _ <- resp.getJobReport.getStatus match {
+        case JobReport.StatusEnum.SUCCEEDED =>
+          for {
+            _ <- logger.info(ctx.loggingCtx)(
+              s"vm ${params.runtime.id} is deleted successfully"
+            )
+          } yield ()
+        case JobReport.StatusEnum.FAILED =>
+          logger.error(s"Wsm deleteVm job failed due to ${resp.getErrorReport.getMessage}") >> F.raiseError[Unit](
+            AzureRuntimeDeletionError(
+              params.runtime.id,
+              params.workspaceId,
+              s"WSM delete VM job failed due to ${resp.getErrorReport.getMessage}"
+            )
+          )
+        case JobReport.StatusEnum.RUNNING =>
+          F.raiseError[Unit](
+            AzureRuntimeDeletionError(
+              params.runtime.id,
+              params.workspaceId,
+              s"WSM delete VM job was not completed within ${config.deleteVmPollConfig.maxAttempts} attempts with ${config.deleteVmPollConfig.interval} delay"
+            )
+          )
+      }
+    } yield ()
+
+  private def monitorDeleteStorageContainer(params: PollStorageContainerParams)(implicit
+    ev: Ask[F, AppContext]
+  ): F[Unit] =
+    for {
+      ctx <- ev.ask
+      wsmApi <- buildWsmControlledResourceApiClient
+      getWsmJobResult = F.delay(
+        wsmApi.getDeleteAzureStorageContainerResult(params.workspaceId.value, params.jobId.value)
+      )
+
+      _ <- F.sleep(
```

Comment:
Same here, I do not think we need a delay because the deletion is so short they used to have a sync endpoint for it